overlord/hookstate/ctlcmd/kmod.go

```diff
@@ -0,0 +1,200 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ctlcmd
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/osutil/kmod"
+	"github.com/snapcore/snapd/overlord/hookstate"
+	"github.com/snapcore/snapd/overlord/ifacestate"
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/snap"
+)
+
+var (
+	shortKmodHelp = i18n.G("Load or unload kernel modules")
+	longKmodHelp  = i18n.G(`
+The kmod command handles loading and unloading of kernel modules.`)
+)
+
+func init() {
+	addCommand("kmod", shortKmodHelp, longKmodHelp, func() command {
+		cmd := &kmodCommand{}
+		cmd.InsertCmd.kmodBaseCommand = cmd
+		cmd.RemoveCmd.kmodBaseCommand = cmd
+		return cmd
+	})
+}
+
+type kmodCommand struct {
+	baseCommand
+	InsertCmd KModInsertCmd `command:"insert" description:"load a kernel module"`
+	RemoveCmd KModRemoveCmd `command:"remove" description:"unload a kernel module"`
```

Comment:
s/InsertCmd/LoadCmd/
s/RemoveCmd/UnloadCmd/ ?

Reply:
I think it should be consistent with the name of the command. I've now picked `insert` and `remove` after looking at `kmod` source code and finding that there are two experimental methods with that name (that is, in the future `kmod insert` and `kmod remove` might replace `modprobe` and `modprobe -r`, respectively), but the spec is still not approved by Samuele, so I'd rather wait for his final call before fixing the names.